private-media-framework/draft-ietf-perc-private-media-framework.md

```diff
@@ -487,10 +485,34 @@ Key Distributor and endpoints will also be used to convey HBH key
 information from the Key Distributor to the Media Distributor, so no
 additional protocol or interface is required.
 
+In establishing the DTLS association between endpoints and the
+Key Distributor, the endpoint acts as the DTLS client and the
+Key Distributor acts as the DTLS server.  The Key Encryption Key (KEK)
+(i.e., EKT Key) is conveyed by the Key Distributor over the DTLS
+association to endpoints via procedures defined in PERC EKT
+[I-D.ietf-perc-srtp-ekt-diet] via the EKTKey message.  Endpoints do not
+send EKTKey messages over the DTLS association and the Key Distributor
+**MUST** silently discard any EKTKey messages received from an endpoint.
+
+Note that following DTLS-SRTP procedures for the [@!I-D.ietf-perc-double]
+cipher, the endpoint will generate both E2E and HBH encryption keys
+and salt values.  Endpoints **MAY** use the DTLS-SRTP generated E2E key
+or **MAY** generate different E2E keys.  In either case, the generated SRTP
+master salt for E2E encryption **MUST** be replaced with the salt value
+provided by the Key Distributor via the EKTKey message.  That is because
+every endpoint in the conference uses the same SRTP master salt.  The
+endpoint only transmits its SRTP master key (not the salt) to other
+endpoints in RTP/RTCP packets per [I-D.ietf-perc-srtp-ekt-diet].
```

Comment:
Suggest adding:

```
When an endpoint sends an EKT tag, it **MUST** send a half-length key, so that only the E2E portion of the key is replaced.
```

Reply:
That can be a little confusing when you start talking about halves.  How about "The
endpoint only transmits the SRTP master key used for E2E encryption..." in that line above?

Reply:
So that last sentence would read:
"The endpoint only transmits the SRTP master key (not the salt) used for E2E encryption to other endpoints in RTP/RTCP packets per [I-D.ietf-perc-srtp-ekt-diet]."

Reply:
No, we need to be concrete here in order for this all to hang together:
* In -double the first half of the key is the E2E key
* In -ekt, you replace the first N bits of the SRTP key with what you get
* Thus in -framework, we need specify that the first half of the octets of the key are sent in the EKT tag

Reply:
How is "The endpoint only transmits the SRTP master key (not the salt) used for E2E encryption" not concrete?  As an implementer, would you know what key goes there?  Which part of the double key is for E2E or HBH is discussion for Double.